Sources/CacheAdvance/EncodableInteger.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  Created by Dan Federman on 12/3/19.
+//  Copyright © 2019 Dan Federman.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+/// A protocol representing an integer that can be encoded as data.
+protocol EncodableInteger {
```

Comment:
Should this protocol be bound to [ExpressibleByIntegerLiteral](https://developer.apple.com/documentation/swift/expressiblebyintegerliteral)?

Reply:
Huh, how does this compile when we bind this protocol to `ExpressibleByIntegerLiteral` if neither `UInt32` or `UInt64` are listed as conforming types? What am I missing?
![AC5F3721-B8C5-4671-A85E-E796D7BC544B](https://user-images.githubusercontent.com/1791049/70379706-b8c8d100-18e4-11ea-807f-5d9d69ab7669.png)

Reply:
The key is `Numeric`, which is a protocol that inherits from this protocol and also is conformed to (indirectly) by `UInt32` and `UInt64`

Reply:
Yep! Wow that inheritance chain goes deep!

Reply:
Indeed! Thanks for asking. Wouldn’t have found `FixedWidthInteger` if you hadn’t